build/dsc/templates/dsc_type.rb.erb

```diff
@@ -130,9 +176,8 @@ Puppet::Type.newtype(:dsc_<%= resource.friendlyname.downcase %>) do
       <%= type_map.to_s %>
     end
 <%    end -%>
-<%    if property.description -%>
-    desc "<%= property.description %>"
-<%    end -%>
+    desc "<%= property.name %><% if property.description -%> - <%= property.description %><% end %>
+      <% if property.values -%>Valid values are <%= property.values.join(', ') %><% end -%>"
```

Comment:
The only nit here is if there isn't a range of values, there is a newline plus empty space here. Example:

```
    desc "Name
      "
```